devicetypes/keen-home/keen-home-smart-vent.src/keen-home-smart-vent.groovy

```diff
@@ -51,423 +46,137 @@ metadata {
+                            [value: 40, color: "#153591"],
+                            [value: 44, color: "#1e9cbb"],
+                            [value: 59, color: "#90d2a7"],
+                            [value: 74, color: "#44b621"],
+                            [value: 84, color: "#f1d801"],
+                            [value: 95, color: "#d04e00"],
+                            [value: 96, color: "#bc2323"]
+                    ]
         }
         valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat") {
             state "battery", label: 'Battery \n${currentValue}%', backgroundColor:"#ffffff"
         }
-        valueTile("zigbeeId", "device.zigbeeId", inactiveLabel: true, decoration: "flat") {
-            state "serial", label:'${currentValue}', backgroundColor:"#ffffff"
-        }
         main "switch"
         details(["switch","refresh","temperature","levelSliderControl","battery"])
     }
 }
 
-/**** PARSE METHODS ****/
+def getPRESSURE_MEASUREMENT_CLUSTER() {0x0403}
+def getMFG_CODE() {0x115B}
+
 def parse(String description) {
     log.debug "description: $description"
-
-    Map map = [:]
-    if (description?.startsWith('catchall:')) {
-        map = parseCatchAllMessage(description)
-    }
-    else if (description?.startsWith('read attr -')) {
-        map = parseReportAttributeMessage(description)
-    }
-    else if (description?.startsWith('temperature: ') || description?.startsWith('humidity: ')) {
-        map = parseCustomMessage(description)
-    }
-    else if (description?.startsWith('on/off: ')) {
-        map = parseOnOffMessage(description)
-    }
-
-    log.debug "Parse returned $map"
-    return map ? createEvent(map) : null
-}
-
-private Map parseCatchAllMessage(String description) {
-    log.debug "parseCatchAllMessage"
-
-    def cluster = zigbee.parse(description)
-    log.debug "cluster: ${cluster}"
-    if (shouldProcessMessage(cluster)) {
-        log.debug "processing message"
-        switch(cluster.clusterId) {
-            case 0x0001:
-                return makeBatteryResult(cluster.data.last())
-                break
-
-            case 0x0402:
-                // temp is last 2 data values. reverse to swap endian
-                String temp = cluster.data[-2..-1].reverse().collect { cluster.hex1(it) }.join()
-                def value = convertTemperatureHex(temp)
-                return makeTemperatureResult(value)
-                break
-
-            case 0x0006:
-                return makeOnOffResult(cluster.data[-1])
-                break
+    def event = zigbee.getEvent(description)
+    if (!event) {
+        Map descMap = zigbee.parseDescriptionAsMap(description)
+        if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.attrInt == 0x0021) {
+            event = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+        } else if (descMap?.clusterInt == PRESSURE_MANAGEMENT_CLUSTER && descMap.attrInt == 0x0020) {
```

Comment:
Attribute ID 0x0020 isn't defined in ZCL8. Is this a manufacturer specific attribute?

Reply:
Nevermind, I see that it is. Probably worth adding a comment that this is a manufacturer specific cluster.